openhands/runtime/impl/remote/remote_runtime.py

```diff
@@ -460,13 +461,18 @@ def list_files(self, path: str | None = None) -> list[str]:
         assert isinstance(response_json, list)
         return response_json
 
-    def copy_from(self, path: str) -> bytes:
+    def copy_from(self, path: str) -> Path:
         """Zip all files in the sandbox and return as a stream of bytes."""
         params = {'path': path}
         response = self._send_request(
             'GET',
             f'{self.runtime_url}/download_files',
             params=params,
+            stream=True,
             timeout=30,
         )
-        return response.content
+        temp_file = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
Guessing this gives a good unique name, like a UUID?

Reply:
Even better than UUID - It yields a 128 bit unique id but in base 36 instead of base 16 - which is shorter!